### ISSUE TITLE
Data not loading in the dashboard

### ISSUE THREAD

**Issue by itsP33t** [G4brym/R2-Explorer#48]
I've migrated from 4 months old version to the latest and my data is not loading. I don't know why. I lost the project files, so i just created new project with the same name and deployed it. I want to add these two buckets:

![](https://cdn.p33t.net/XJYFXTMLXI.png)

So i added this into the `wrangler.toml`

```toml
[[r2_buckets]]
binding = 'p33t-cdni'
bucket_name = 'p33t-cdni'
preview_bucket_name = 'p33t-cdni/'

[[r2_buckets]]
binding = 'verse'
bucket_name = 'verse'
preview_bucket_name = 'verse/'
```

i don't know what to put in the `preview_bucket_name`, i tried the same like `bucket_name` and `/` but nothing is showing any data. When i upload anything to it, it uploads but is not showing in the exporer, in the cloudflare dashboard, it creates a folder `/` and uploads the file there. I'm also receiving a http 400 error:

```
Uncaught (in promise) 
Object { message: "Request failed with status code 400", name: "AxiosError", code: "ERR_BAD_REQUEST", config: {…}, request: XMLHttpRequest, response: {…}, stack: "" }
​
code: "ERR_BAD_REQUEST"
​
config: Object { timeout: 0, xsrfCookieName: "XSRF-TOKEN", xsrfHeaderName: "X-XSRF-TOKEN", … }
​​
adapter: function exports(e)
​​
data: undefined
​​
env: Object { FormData: null }
​​
headers: Object { Accept: "application/json, text/plain, */*", Authorization: "Basic (CENSORED)" }
​​
maxBodyLength: -1
​​
maxContentLength: -1
​​
method: "get"
​​
params: Object { delimiter: "/", prefix: "", cursor: null }
​​
timeout: 0
​​
transformRequest: Array [ transformRequest(e, t) ]
​​
transformResponse: Array [ transformResponse(e) ]
​​
transitional: Object { silentJSONParsing: true, forcedJSONParsing: true, clarifyTimeoutError: false }
​​
url: "/api/buckets/verse?include=customMetadata&include=httpMetadata"
​​
validateStatus: function validateStatus(e)
​​
xsrfCookieName: "XSRF-TOKEN"
​​
xsrfHeaderName: "X-XSRF-TOKEN"
​​
<prototype>: Object { … }
​
message: "Request failed with status code 400"
​
name: "AxiosError"
​
request: XMLHttpRequest { readyState: 4, timeout: 0, withCredentials: false, … }
​
response: Object { data: {…}, status: 400, statusText: "Bad Request", … }
​
stack: ""
​
<prototype>: Object { constructor: i(e, t, r, n, i), toJSON: toJSON(), stack: "", … }
```

**Comment by G4brym** [comment]
Hey there, are you trying to run r2-explorer locally, with wrangler dev?
The `preview_bucket_name` value should be the same as `bucket_name`

**Comment by itsP33t** [comment]
> Hey there, are you trying to run r2-explorer locally, with wrangler dev? The `preview_bucket_name` value should be the same as `bucket_name`

Hey @G4brym i tried running it locally and also just deploying it. The first time trying, i had preview_bucket_name and bucket_name values same, but still this error

**Comment by canuc** [comment]
Yeah if you update the listBucketObjects http request to not include the prefix then the request works, otherwise it always returns 400.

**Comment by canuc** [comment]
needs to fix with a line:

```
prefix: Query(z.string().nullable().optional() ...
```

youll notice the optional + nullable

**Comment by G4brym** [comment]
yeah, you are right @canuc 
this was caused by one of the last itty-router-openapi releases, and i was only able to replicate this issue by updating it

**Comment by G4brym** [comment]
for now as a quickfix i've locked the itty-router-openapi dependency in the older version
please update to the latest r2-explorer version and check if everything works now

**Comment by canuc** [comment]
PLEASE DO NOT USE THIS, IT LEAKS AUTH CREDS TO https://api.r2explorer.dev/api/releases/latest from the FE. this is not good and means creator will be able to hijack accounts.

**Comment by G4brym** [comment]
@canuc as i said in #49 
If you are concerned with authentication, you should definitely enable Cloudflare Access on your instance [docs here](https://r2explorer.dev/getting-started/security/#authenticating-with-cloudflare-access).

**Comment by G4brym** [comment]
Thanks for reporting this issue, I'm closing this, as the issue has been fixed in the latest release (v1.0.5)
Update your instance with `npm install r2-explorer@latest --save` and then deploy it with `wrangler publish`
Let me know if you find any other bug
